### DISCUSSION TITLE
Error with Build, Linux Noob

### DISCUSSION THREAD

**Discussion by saltbr1nger** [polybar/polybar#2818] [Build Support]
Linux noob here trying to install polybar.  Running Mint 20.3.  Have all dependencies installed.  This is what happens when i try to run the build.sh script.  Not sure what to do from here.

** A build dir already exists (pass -f to replace)

** Setting build options

** Use GCC even if Clang is installed ----------------------------- [y/N]: y
** Include support for "internal/i3" (requires i3) ---------------- [y/N]: y
** Include support for "internal/alsa" (requires alsalib) --------- [y/N]: y
** Include support for "internal/pulseaudio" (requires libpulse) -- [y/N]: y
** Include support for "internal/network" (requires libnl/libiw) -- [y/N]: y
** Include support for "internal/mpd" (requires libmpdclient) ----- [y/N]: y
** Include support for "internal/github" (requires libcurl) ------- [y/N]: y
** Build "polybar-msg" used to send ipc messages ------------------ [y/N]: y
** Parallelize the build using make -j6 --------------------------- [y/N]: y
** Executing cmake command

-- Could not detect version with git, falling back to built-in version information.
-- No build type specified; using Release
-- Couldn't locate ccache, disabling ccache...
-- Using supported compiler GNU-9.4.0
-- Checking for module 'cairo-fc'
--   No package 'cairo-fc' found
CMake Error at /usr/share/cmake-3.16/Modules/FindPkgConfig.cmake:463 (message):
  A required package was not found
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindPkgConfig.cmake:643 (_pkg_check_modules_internal)
  cmake/common/utils.cmake:45 (pkg_check_modules)
  cmake/modules/FindCairoFC.cmake:10 (find_package_impl)
  cmake/libpoly.cmake:49 (find_package)
  CMakeLists.txt:49 (include)

-- Configuring incomplete, errors occurred!
See also "/home/defcon3/Downloads/polybar-3.6.3/build/CMakeFiles/CMakeOutput.log".
See also "/home/defcon3/Downloads/polybar-3.6.3/build/CMakeFiles/CMakeError.log".
** Failed to generate build... read output to get a hint of what went wrong

**Comment by patrick96** [comment]
Is the `libcairo2-dev` package installed?

**Reply by saltbr1nger** [reply]
I don't think so.  Installed it and tried again, but error again with the build.

Actually upon second look, the error is slightly different now:

-- Could not detect version with git, falling back to built-in version information.
-- No build type specified; using Release
-- Couldn't locate ccache, disabling ccache...
-- Using supported compiler GNU-9.4.0
-- Checking for module 'libuv'
--   No package 'libuv' found
CMake Error at /usr/share/cmake-3.16/Modules/FindPkgConfig.cmake:463 (message):
  A required package was not found
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindPkgConfig.cmake:643 (_pkg_check_modules_internal)
  cmake/common/utils.cmake:45 (pkg_check_modules)
  cmake/modules/FindLibUV.cmake:9 (find_package_impl)
  cmake/libpoly.cmake:51 (find_package)
  CMakeLists.txt:49 (include)

**Reply by patrick96** [reply]
Did you install all the dependencies listed on the wiki? Because now you're missing `libuv1-dev`

**Reply by saltbr1nger** [reply]
Yea you're right, I missed some dependencies.  I installed them all this time though, got further into the build, but still errored out.  This is the latest error I'm getting now:

ImportError: cannot import name 'gcd' from 'fractions' (/home/defcon3/anaconda3/lib/python3.9/fractions.py)
make[2]: *** [lib/xpp/CMakeFiles/xpp.dir/build.make:70: lib/xpp/generated-sources/include/xpp/proto/xkb.hpp] Error 1
make[2]: *** Deleting file 'lib/xpp/generated-sources/include/xpp/proto/xkb.hpp'
make[1]: *** [CMakeFiles/Makefile2:492: lib/xpp/CMakeFiles/xpp.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  8%] Built target doc_html
[  8%] Built target doc_man
make: *** [Makefile:130: all] Error 2
** Failed to build project

**Reply by patrick96** [reply]
Hmm that looks like the same issue as in #2293. What is the version of your `python3-xcbgen` package?

**Reply by patrick96** [reply]
Alternatively, what is the python version installed from the parrot repos? You are currently using python3.9 from anaconda. You most likely will need to use python3.8 because you have an older python3-xcbgen package.

**Reply by saltbr1nger** [reply]
python3-xcbgen is 1.14-2

re: python version from parrot repos, idk.  this is like my 2nd week on Linux; i thought my python version was the same as my anaconda installation (3.9).  How do I check this?

**Reply by patrick96** [reply]
I'm pretty sure the linux mint python version is 3.8 since it's based on Ubuntu 20.04. So in your case you need to deactivate conda while installing polybar

**Reply by saltbr1nger** [reply]
yep, i go `conda deactivate` but still get the same error.  gonna try  tweaking the align.py file like someone mentioned in the #2293 thread, and if that doesnt work, renaming the fractions.py file or whatever.  pray for me

edit: it worked.  https://github.com/polybar/polybar/issues/2293#issuecomment-1068920777
